show/hide UI components for offline access

sheralim012 opened the pull request:
Closes #881 

- show/hide following UI components in online/offline mode
  - search input and search button
  - comments
    - posting
    - liking comment
    - replying comment
  - chatbot buttons
  - offline app buttons
  - accounts
    - login
    - logout
    - create account
    - change digital pin
  - progressbar
- for questionnaires submit/next button
  - append text "(You cannot submit when offline)"
  - disable the button

geoo89 commented:
For reading progress, the articles that have been read are still being highlighted: 
![image](https://user-images.githubusercontent.com/678199/153363323-af2a7b8a-1b17-4a4e-b8d6-f0d73c89ae77.png)

For the offline button, could we grey it out + make it unclickable and render it with the "offline" message that is specified when you create such a button?

Similarly, could we grey out the chatbot button rather than hiding it altogether?

Report button still appears for comments.

Questionnaire submit buttons still seem to appear as normal -- did I resolve the merge conflicts incorrectly?

Logout button appears on the profile page. Maybe this needs discussion and we can defer it to later, also see https://github.com/unicef/iogt/issues/514#issuecomment-1020297835 
When the site is cached, are the pages loaded as if the user was logged in (when they are logged in)?

sheralim012 commented:
> The lock is now missing from the "change password" button when I'm online.

could you try `python manage.py collectstatic` 

> The chatbot button now redirects me when I'm offline, but doesn't do anything when I'm online. This should be reversed.

somehow it was removed in the last commit, added back the handlers for it

> The report button now does not appear at all, also when online.

fixed

> Could you merge/rebase develop again, now that it has passing tests, to check whether this PR makes any tests fail?

done
